scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3114,35 +3118,56 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) {
         name match {
           case q: Quasi if endTparamQuasi => q.become[Type.Param]
           case _ =>
-            val tparams = typeParamClauseOpt(ownerIsType = true, ctxBoundsAllowed = false)
+            val tparams = typeParamClauseOpt(
+              ownerIsType = true,
+              ctxBoundsAllowed = dialect.allowImprovedBoundsAndGivens
+            )
             val tbounds = typeBounds()
             val vbounds = new ListBuffer[Type]
             val cbounds = new ListBuffer[Type]
             if (ctxBoundsAllowed) {
               @inline
-              def getBound(): Type = {
-                next()
-                currToken match {
-                  case t: Ellipsis => ellipsis[Type](t, 1)
-                  case _ => typ()
-                }
+              def getBound(allowBoundsAlias: Boolean = false): Type = currToken match {
+                case t: Ellipsis => ellipsis[Type](t, 1)
+                case _ => contextBoundOrAlias(allowBoundsAlias)
               }
-              if (at[Viewbound]) {
+              if (acceptOpt[Viewbound]) {
```

Comment:
looking at the code now, and in the context of parsing resiliency discussions we've had, I would:
- get rid of the `ctxBoundsAllowed` parameter to this method
- not add the new dialect flag

either there's a ViewBound and/or Colon token, in which case we'll add the bounds, or there isn't. these tokens can't be parsed differently based on either of those two flags above.

i'd even deprecate and ignore the `allowViewBounds` dialect flag and remove the error just below.

Reply:
I would keep the dialect flag since `A as otherName` would be an infix type before, so this would change parsing for Scala 2 even

Reply:
ah, logical :)

Reply:
I removed allowViewBounds as it's not really useful

Reply:
to answer your question: it still looks to me that the old parameter is not needed, only the dialect flag to distinguish between alias and type infix.

Reply:
Right, makes sense! I removed ctxBoundsAllowed param